gprotorch/kernels/gnn_kernels/test_hu_iclr_2020.py

```diff
@@ -0,0 +1,357 @@
+"""
+Reformatted and cleaned up GNN modules from Hu et al.
+Strategies for Pre-training Graph Neural Networks. ICLR 2020
+(https://github.com/snap-stanford/pretrain-gnns)
+
+Author: Leo Klarner (https://github.com/leojklarner), April 2022
+"""
+
+import os
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from rdkit import Chem
+from torch_geometric.data import Data
+from torch_geometric.nn import MessagePassing
+from torch_geometric.utils import add_self_loops
+from torch_scatter import scatter_add
+
+# define the allowable node and edge labels as used in Hu et al.
+allowable_features = {
+    "possible_atomic_num_list": list(range(1, 119)),
+    "possible_chirality_list": [
+        Chem.rdchem.ChiralType.CHI_UNSPECIFIED,
+        Chem.rdchem.ChiralType.CHI_TETRAHEDRAL_CW,
+        Chem.rdchem.ChiralType.CHI_TETRAHEDRAL_CCW,
+        Chem.rdchem.ChiralType.CHI_OTHER,
+    ],
+    "possible_bonds": [
+        Chem.rdchem.BondType.SINGLE,
+        Chem.rdchem.BondType.DOUBLE,
+        Chem.rdchem.BondType.TRIPLE,
+        Chem.rdchem.BondType.AROMATIC,
+    ],
+    # (E)/(Z) double bond stereo information
+    "possible_bond_dirs": [
+        Chem.rdchem.BondDir.NONE,
+        Chem.rdchem.BondDir.ENDUPRIGHT,
+        Chem.rdchem.BondDir.ENDDOWNRIGHT,
+    ],
+}
+
+# define additional tokens for label masking
+num_atom_type = 120  # including the extra mask tokens
+num_chirality_tag = 3
+
+num_bond_type = (
+    6  # including aromatic and self-loop edge, and extra masked tokens
+)
+num_bond_direction = 3
+self_loop_token = 4  # bond type for self-loop edge
+masked_bond_token = 5  # bond type for masked edges
+
+
+def mol_to_pyg(mol):
+    """
+    A featuriser that accepts an rdkit mol instance and
+    converts it to a PyTorch Geometric data object that
+    is compatible with the GNN modules below.
+
+    Args:
+        mol: rdkit mol object
+
+    Returns: PyTorch Geometric data object
+
+    """
+
+    # derive atom features: atomic number + chirality tag
```

Comment:
What does the chirality tag look like out of interest?

Reply:
Can it help to enrich the atom features with other properties?

Reply:
For atoms with four unique neighbours it's whether they are ordered clockwise (- form) or counter-clockwise (+ form), see https://en.wikipedia.org/wiki/Chirality_(chemistry).
Yes, including additional atom features would probably lead to improved performance, but you'd have to retrain the self-supervised initialisations.